models/ExperimentFull.ts

```diff
@@ -185,7 +185,7 @@ export class ExperimentFull implements ApiDataSource {
   /**
    * Return the primary metric assignment ID for this experiment if one exists.
    */
-  getPrimaryMetricAssignmentId(): number | undefined {
-    return this.metricAssignments.find((metricAssignment) => metricAssignment.isPrimary)?.metricAssignmentId
+  getPrimaryMetricAssignmentId(): number | null {
+    return this.metricAssignments.find((metricAssignment) => metricAssignment.isPrimary)?.metricAssignmentId || null
```

Comment:
```suggestion
    return this.metricAssignments.find((metricAssignment) => metricAssignment.isPrimary)?.metricAssignmentId ?? null
```
I don't know if `metricAssignmentId` could be `0`, but if it is then we don't want to convert it to `null`.

The `??` is the nullish_coalescing_operator. It's fairly new to JS (supported by TS). See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator for details.

Reply:
It can't be zero, but no harm in using `??`. Thanks! :slightly_smiling_face: